ovn-tester/ovn_utils.py

```diff
@@ -194,60 +240,199 @@ def unbind_vm_port(self, lport):
         self.run(f'ip netns del {lport.name}', prefix='')
 
 
-class OvnNbctl:
-    def __init__(self, sb):
-        self.sb = sb
-        self.socket = ""
+# We have to subclass the base Transaction for NB in order to facilitate the
+# "sync" command. This is heavily based on ovsdbapp's OvsVsctlTransaction class
+# but with some NB-specific modifications, and removal of some OVS-specific
+# assumptions.
+class NBTransaction(transaction.Transaction):
+    def __init__(
+        self,
+        api,
+        ovsdb_connection,
+        timeout=None,
+        check_error=False,
+        log_errors=True,
+        wait_type=None,
+        **kwargs,
+    ):
+        super(NBTransaction, self).__init__(
+            api,
+            ovsdb_connection,
+            timeout=timeout,
+            check_error=check_error,
+            log_errors=log_errors,
+            **kwargs,
+        )
+        self.wait_type = wait_type.lower() if wait_type else None
+        if (
+            self.wait_type
+            and self.wait_type != "sb"
+            and self.wait_type != "hv"
+        ):
+            log.warning(f"Unrecognized wait type {self.wait_type}. Ignoring")
+            self.wait_type = None
+
+    def pre_commit(self, txn):
+        if self.wait_type:
+            self.api._nb.increment('nb_cfg')
+
+    def post_commit(self, txn):
+        super().post_commit(txn)
+        try:
+            self.do_post_commit(txn)
+        except ovsdbapp_exceptions.TimeoutException:
+            log.exception("Transaction timed out")
+
+    def do_post_commit(self, txn):
+        if not self.wait_type:
+            return
+
+        next_cfg = txn.get_increment_new_value()
+        while not self.timeout_exceeded():
+            self.api.idl.run()
+            if self.nb_has_completed(next_cfg):
+                break
+            self.ovsdb_connection.poller.timer_wait(
+                self.time_remaining() * 1000
+            )
+            self.api.idl.wait(self.ovsdb_connection.poller)
+            self.ovsdb_connection.poller.block()
+        else:
+            raise ovsdbapp_exceptions.TimeoutException(
+                commands=self.commands,
+                timeout=self.timeout,
+                cause='nbctl transaction did not end',
+            )
 
-    def __del__(self):
-        # FIXME: the SSH connection might have already been closed here..
-        # self.stop_daemon()
-        pass
+    def nb_has_completed(self, next_cfg):
+        if not self.wait_type:
+            return True
+        elif self.wait_type == "sb":
+            cur_cfg = self.api._nb.sb_cfg
+        else:  # self.wait_type == "hv":
+            cur_cfg = min(self.api._nb.sb_cfg, self.api._nb.hv_cfg)
```

Comment:
I was about to mention about how python doesn't have issues with rollover like C. But since the DB server is likely limiting the value to LLONG_MAX, we need to deal with that in our python code as well.

And to answer the first question, I copied the behavior from `ovn-nbctl.c` : 
````
int64_t cur_cfg = (wait_type == NBCTL_WAIT_SB
                   ? nb->sb_cfg
                   : MIN(nb->sb_cfg, nb->hv_cfg));
````

Reply:
Oh, oops, this was introduced to fix something I had reported :)
https://github.com/ovn-org/ovn/commit/cd3c0e1c2ab3a230965987e6c2be6cf8218226f6

It's to cover the corner case when there's no active chassis in the cluster.

For "dealing" with the overflow we can maybe check the value of `nb_cfg` incrementing it in `NBTransaction.pre_commit()`.

Reply:
I just pushed new changes and realized I did not address this. I'll push another update in just a bit.